### PR TITLE
Caching Dependencies to speed up workflows

### PR DESCRIPTION
**Pull request by jongwooo**
Signed-off-by: jongwooo <han980817@gmail.com>
Issue #605 

## Description
I updated workflows to apply **dependency caching**.

### About caching workflow dependencies
Jobs on GitHub-hosted runners start in a clean virtual environment and **must download dependencies each time**, causing increased network utilization, longer runtime, and increased cost. To help speed up the time it takes to recreate files like dependencies, GitHub can cache files that frequently use in workflows.

### Solutions
Java projects can run faster on GitHub Actions by enabling dependency caching on the [setup-java](https://github.com/actions/setup-java) action. `setup-java` supports caching for both Gradle and Maven projects.

The following example enables caching for a Java project with Gradle:

```yaml
steps:
- uses: actions/checkout@v3
- uses: actions/setup-java@v3
  with:
    java-version: '11'
    distribution: 'temurin'
    cache: 'gradle'
- run: ./gradlew build
```

> It’s literally a one line change to pass the`cache: gradle` input parameter.

## References
[actions/setup-java#caching-packages-dependencies](https://github.com/actions/setup-java#caching-packages-dependencies)
[How to build Gradle projects with GitHub Actions](https://tomgregory.com/build-gradle-projects-with-github-actions/#5_Enable_Gradle_caching)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by soimkim** [comment]
@jongwooo , 
How about changing distribution from adopt to temulin?

> NOTE: Adopt OpenJDK got moved to Eclipse Temurin and won't be updated anymore. It is highly recommended to migrate workflows from adopt to temurin to keep receiving software and security updates. See more details in the [Good-bye AdoptOpenJDK post](https://blog.adoptopenjdk.net/2021/08/goodbye-adoptopenjdk-hello-adoptium/).

**Comment by jongwooo** [comment]
@soimkim Great idea!

**Comment by riyenas0925** [comment]
@jongwooo 
You need to fix it to pass the test.

image

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -73,14 +73,19 @@ jobs:
         with:
           time: '30s'
 
+      - name: Grant permission for db data
```
Comment (jongwooo) [inline .github/workflows/pull-request.yml:76]: @riyenas0925 I updated the workflow to **grant permission** for db data. I hope this change can solve the problem.

Comment (riyenas0925) [reply .github/workflows/pull-request.yml:76]: LGTM 👍

Comment (soimkim) [reply .github/workflows/pull-request.yml:76]: Shouldn't we add permission change to `db/data` in` release-publish.yml` as well?

Comment (jongwooo) [reply .github/workflows/pull-request.yml:76]: The `db/data` is created by running `docker:db`. There was no related step in the `release-publish.yml`, so I excluded it.

Comment (soimkim) [reply .github/workflows/pull-request.yml:76]: @jongwooo , I mean, even in release-publish, there will be an error saying that it does not have db/data permission when building, so we need to add the db/data permission change to release-publish.yml as well.

Comment (soimkim) [reply .github/workflows/pull-request.yml:76]: Oh, it's that release doesn't run docker. now i understand

**Pull request merged**